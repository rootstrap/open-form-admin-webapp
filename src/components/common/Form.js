import styled, { css } from 'styled-components';
import { Form } from 'formik';

export const formStyles = css`
  display: flex;
  flex-direction: column;
  margin: 1em 0;
  width: 50%;
  align-self: center;
  max-width: 730px;
`;

export default styled(Form)`
  ${formStyles}
`;
