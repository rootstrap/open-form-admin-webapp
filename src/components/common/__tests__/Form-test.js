import React from 'react';
import TestRenderer from 'react-test-renderer';
import styled from 'styled-components';

import { formStyles } from '../Form';

/**
 * doing this because we don't wanna test formik internals
 * testing this way, it will let us know if something went wrong with styles
 */
const StyledForm = styled.form`
  ${formStyles}
`;

it('renders correctly', () => {
  const testRenderer = TestRenderer.create(<StyledForm>Form</StyledForm>);
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
