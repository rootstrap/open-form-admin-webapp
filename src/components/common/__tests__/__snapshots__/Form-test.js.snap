// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1em;
}

<form
  action="#"
  className="c0"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    type="text"
  />
</form>
`;
